app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -24,19 +35,79 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
+             *  - - - Diffing rules: value from currentFilters takes priority over value from search. this is because
+             *  the user will always have selected the value after they select a saved search,meaning they want to change it.
+             *  - - - Diffing rules: if the value is equal, just ignore it/send empty value.
+             *  - - - Diffing rules: return the _difference if the value is not equal, because we will want to show for instance:
+             *  saved search: tag id 1 + filters tag id 2 (so it's not just ignoring the arrays)
+             */
+            $scope.userCanUpdateSavedSearch = false;
+            $scope.uiFilters = currentFilters;
+
+            if ($scope.savedSearch) {
+                /**
+                 * to handle removal correctly we need to make sure we take currentFilters (which has up to date info) into account,
+                 * because that is where our savedsearch filters will stop being represented when we remove them.
+                 * If there's a key in our current filters that is in the saved search but is not in the active filters at this point,
+                 * it is because it was removed (since before saved search gets assigned, they are all assigned to the filters)
+                 * that means we have to remove it from the saved search.
+                 **/
+                $scope.savedSearch.filter = PostFilters.cleanRemovedValuesFromObject(filters, PostFilters.getUIActiveFilters($scope.savedSearch.filter));
+                /**
+                 * Add back in savedSearch.filter if an originally saved search filter is removed+added back
+                 */
+                $scope.savedSearch.filter = PostFilters.getUIActiveFilters(PostFilters.addIfCurrentObjectMatchesOriginal(PostFilters.getUIActiveFilters($scope.savedSearch.filter), PostFilters.getUIActiveFilters(originalSavedSearch.filter), PostFilters.getUIActiveFilters($scope.uiFilters)));
+                var savedSearchFiltersChanged = _.filter(PostFilters.getUIActiveFilters(originalSavedSearch.filter), function (obj, key) {
+                    if (!_.isArray(obj)) {
+                        return $scope.savedSearch.filter[key] !== obj;
+                    } else {
+                        return _.difference($scope.savedSearch.filter[key], obj).length !== 0 || _.difference(obj, $scope.savedSearch.filter[key]).length !== 0 ;
+                    }
+                }).length > 0 ;
+                var filtersDifferentToSavedSearch = !_.isEqual($scope.savedSearch.filter, filters);
+                $scope.userCanUpdateSavedSearch = _.contains($scope.savedSearch.allowed_privileges, 'update') && (savedSearchFiltersChanged || filtersDifferentToSavedSearch);
+                // remove values that are in the saved search frmo the uifilters.
+                _.each($scope.uiFilters, function (value, key) {
```

Comment:
FYI I moved this to a service and made it more sane . Pushing today after testing everything again